src/FastIntegerDivide.cpp

```diff
@@ -10,17 +10,28 @@ using namespace Halide::Internal::IntegerDivision;
 
 namespace {
 
+int shift_for_denominator(uint32_t d) {
+    return 31 - __builtin_clz(d - 1);
+}
+
+Expr shift_for_denominator(const Expr &d) {
+    internal_assert(d.type().element_of() == UInt(8));
+    return 7 - count_leading_zeros(d - 1);
+}
+
 Buffer<uint8_t> integer_divide_table_u8() {
     static std::mutex initialize_lock;
     std::lock_guard<std::mutex> lock_guard(initialize_lock);
     {
-        static Buffer<uint8_t> im(256, 2);
+        static Buffer<uint8_t> im(256);
```

Comment:
Unrelated: since C++11 onwards default to promising thread-safe initialization of statics, we could rewrite this to be a bit terser like so:

```
Buffer<uint8_t> integer_divide_table_u8() {
    static Buffer<uint8_t> im(256) = []() {
        Buffer<uint8_t> im(256);
        for (uint32_t i = 0; i < 256; i++) {
            im(i) = table_runtime_u8[i][2];
            if (i > 1) {
                internal_assert(table_runtime_u8[i][3] == shift_for_denominator(i));
            }
        }
        return im;
    }();

    return im;
}
```

Reply:
Good point. Will fix.